### PR TITLE
XW-3206 | Correct response status codes in Express

### PR DESCRIPTION
**Pull request by kvas-damian**
## Links

* http://wikia-inc.atlassian.net/browse/XW-3206

## Description

Previously we were using incorrect statusCode that we finally served to client. In this PR on error we get highest statusCode and serve it to browser.

## Reviewers

@rogatty

### CODE REVIEW

File: server/app/server.js [hunk 1]
```diff
@@ -53,14 +54,14 @@ const server = new FastBootAppServer({
 	afterMiddleware: (app) => {
 		app.use((err, req, res, next) => {
 			if (err) {
-				// It should never get here
-				// In theory, the Ember app handles all server errors
-				const level = levelFn(res.statusCode, err);
+				// Handle errors that don't go to FastBoot, like Bad Request etc.
+				const statusCode = Math.max(res.statusCode, err.statusCode);
```
Comment (rogatty) [inline server/app/server.js]: Can we have a situation where there is no `res.statusCode` or `err.statusCode`?

```
> var x = {};
> Math.max(10, x.y)
< NaN
```

File: server/app/server.js [hunk 2]
```diff
@@ -53,14 +54,14 @@ const server = new FastBootAppServer({
...
+				const level = levelFn(statusCode);
 				const logFn = req.log[level].bind(req.log);
 
 				logFn(err);
 
-				res.send('Server error');
+				res.sendStatus(statusCode);
```
Comment (rogatty) [inline server/app/server.js:64]: Does it send a blank page now or a stack trace?

Comment (kvas-damian) [reply server/app/server.js:64]: are you referring to Ember or express stack trace?

`sendStatus()` sends given status code with default message:
https://expressjs.com/en/api.html#res.sendStatus

Comment (rogatty) [reply server/app/server.js:64]: Oh, the default message is :+1:

File: server/app/server.js (hunk 1 above)
Comment (kvas-damian) [reply server/app/server.js]: I didn't found such case. Do you think we fallback to `500` in such cache?

Comment (rogatty) [reply server/app/server.js]: How about `Math.max(parseInt(res.statusCode, 10), parseInt(err.statusCode, 10)) || 500`?

**Review by rogatty — APPROVED** [review]

**Pull request merged**